bot/utils/messages.py

```diff
@@ -164,6 +165,42 @@ async def send_attachments(
     return urls
 
 
+async def count_unique_users_reaction(
+        message: discord.Message,
+        predicate: Callable[[Reaction], bool] = lambda _: True,
+        count_bots: bool = True
+) -> int:
+    """
+    Count the amount of unique users who reacted to the message.
+
+    A predicate function can be passed to check if this reaction should be counted, along with a count_bot flag.
+    """
+    unique_users = set()
+
+    for reaction in message.reactions:
+        if predicate(reaction):
+            async for user in reaction.users():
+                if count_bots or not user.bot:
+                    unique_users.add(user.id)
+
+    return len(unique_users)
```

Comment:
Since you've written this, I wonder if the duckpond cog can be refactored to use it?

Reply:
Probably, although that would be pretty much out of scope for this PR wouldn't it?

Reply:
I'll still approve regardless of this, but I try to apply relevant changes whenever I add a utility function like that. No one will think to PR it separately afterwards.